packages/react/src/ProgressivelyProvider.tsx

```diff
@@ -36,17 +36,20 @@ export const ProgressivelyProvider = ({
 
     setTrackFn(() => sdk.track);
 
-    sdk
-      .loadFlags({ ctrl })
-      .then((res) => {
-        sdk.onFlagUpdate(
-          setFlags,
-          res.response.headers.get("X-progressively-id")
-        );
-        setFlags(res.flags);
-        setIsLoading(false);
-      })
-      .catch(setError);
+    sdk.loadFlags({ ctrl }).then((res) => {
+      sdk.onFlagUpdate(
+        setFlags,
+        res.response.headers.get("X-progressively-id")
+      );
+      setFlags(res.flags);
+
+      setStatus("success");
+
+      if (res.error) {
+        setStatus("failure");
+        setError(res.error.toString());
```

Comment:
Do you mind passing the whole error in the object? People can then decide if they want to get the string value, stack trace or rethrowing to their sentry instance (I dont know)